test/suite/test_rollback_to_stable01.py

```diff
@@ -0,0 +1,116 @@
+
+import time
+from helper import copy_wiredtiger_home
+import unittest, wiredtiger, wttest
+from wtdataset import SimpleDataSet
+from wiredtiger import stat
+
+def timestamp_str(t):
+    return '%x' % t
+
+# test_rollback_to_stable01.py
+# Shared base class used by gc tests.
+class test_rollback_to_stable_base(wttest.WiredTigerTestCase):
+    def large_updates(self, uri, value, ds, nrows, commit_ts):
+        # Update a large number of records.
+        session = self.session
+        cursor = session.open_cursor(uri)
+        for i in range(0, nrows):
+            session.begin_transaction()
+            cursor[ds.key(i)] = value
+            session.commit_transaction('commit_timestamp=' + timestamp_str(commit_ts))
+        cursor.close()
+
+    def large_removes(self, uri, ds, nrows, commit_ts):
+        # Remove a large number of records.
+        session = self.session
+        cursor = session.open_cursor(uri)
+        for i in range(0, nrows):
+            session.begin_transaction()
+            cursor.set_key(i)
+            cursor.remove()
+            session.commit_transaction('commit_timestamp=' + timestamp_str(commit_ts))
+        cursor.close()
+
+    def check(self, check_value, uri, nrows, read_ts):
+        session = self.session
+        session.begin_transaction('read_timestamp=' + timestamp_str(read_ts))
+        cursor = session.open_cursor(uri)
+        count = 0
+        for k, v in cursor:
+            self.assertEqual(v, check_value)
+            count += 1
+        session.rollback_transaction()
+        self.assertEqual(count, nrows)
+
+# Test that rollback to stable clears the remove operation.
+class test_rollback_to_stable01(test_rollback_to_stable_base):
+    # Force a small cache.
+    conn_config = 'cache_size=50MB,log=(enabled),statistics=(all)'
+    session_config = 'isolation=snapshot'
+
+    def test_rollback_to_stable(self):
+        nrows = 10000
+
+        # Create a table without logging.
+        uri = "table:rollback_to_stable01"
+        ds = SimpleDataSet(
+            self, uri, 0, key_format="i", value_format="S", config='log=(enabled=false)')
+        ds.populate()
+
+        # Pin oldest and stable to timestamp 1.
+        self.conn.set_timestamp('oldest_timestamp=' + timestamp_str(1) +
+            ',stable_timestamp=' + timestamp_str(1))
+
+        valuea = "aaaaa" * 100
+        self.large_updates(uri, valuea, ds, nrows, 10)
+
+        # Check that all updates are seen
+        self.check(valuea, uri, nrows, 10)
+
+        # Remove all keys with newer timestamp
+        self.large_removes(uri, ds, nrows, 20)
+
+        # Check that the no keys should be visible
+        self.check(valuea, uri, 0, 20)
+
+        # Pin oldest and stable to timestamp 100.
+        self.conn.set_timestamp('stable_timestamp=' + timestamp_str(10))
+
+        # Checkpoint to ensure that all the updates are flushed to disk.
+        self.session.checkpoint()
+
+        self.conn.rollback_to_stable()
+
```

Comment:
This test probably does not need a blank line after every single line. Please review and remove some.

Reply:
Fixed by removing newlines where they are not necessary.